### ISSUE TITLE
jsii 1.50.0 breaks for monorepos with local dependencies

### ISSUE THREAD

**Issue by mattsb42-aws** [aws/jsii#3319]
### Affected Languages
- [x] `TypeScript` or `Javascript`
- [ ] `Python`
- [ ] `Java`
- [ ] .NET (`C#`, `F#`, ...)
- [ ] `Go`

### General Information
* **JSII Version:** 1.50.0
* **Platform:**     MacOS but irrelevant

### What is the problem?

The dependency resolver added in 1.50.0[1] breaks on local dependencies. Because the `declaration` value is a file URI but the `actualVersion` is whatever version string the dependency package defines, these values can never match. This causes `jsii` to fail when it attempts to compile the package.

[1] https://github.com/aws/jsii/blob/d34881c35a56302ef2a1c3e8a0d19995c3344720/packages/jsii/lib/project-info.ts#L238-L267

### Verbose Log

```
👾 build » compile | jsii --silence-warnings=reserved-word --no-fix-peer-dependencies
Error: Error: Declared dependency on version file:../<REDACTED> of <REDACTED>, but version <REDACTED> was found
```

**Comment by wchaws** [comment]
Encounter the same issue

**Comment by MrArnoldPalmer** [comment]
@mattsb42-aws to clarify this occurs only when using a file path as the dependency version in package.json, but `npm link`ing works okay right? I think we mostly link in our workflow hence why this wasn't caught, looks like you're adding a test in your PR though.

**Comment by mattsb42-aws** [comment]
Correct. Because `declared` is not a version, it will always fail because it can never be a reasonable match for `actualVersion`. We haven't tried `npm link`; I'll look into that, but even if that works for us I still think this is worth fixing.

**Comment by mattsb42-aws** [comment]
It would also affect any other URI.

https://docs.npmjs.com/cli/v7/configuring-npm/package-json#dependencies
